onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h

```diff
@@ -37,6 +39,17 @@ class MKLDNNExecutionProvider : public IExecutionProvider {
   }
 
   virtual std::shared_ptr<KernelRegistry> GetKernelRegistry() const override;
+
+  std::shared_ptr<mkldnn::memory> GetWeightMemory(std::string weightName) {
+    if (weights_mem_map.find(weightName) != weights_mem_map.end())
```

Comment:
How about this:
```
auto iter = weights_mem_map.find(weightName);
if(iter != weights_mem_map.end())
  return *iter;
return nullptr;
```

Reply:
sure.

Reply:
@snnn I am getting the following error at r`eturn *iter; 
`
`Error	C2440	'return': cannot convert from 'std::pair<const _Kty,_Ty>' to 'std::shared_ptr<mkldnn::memory>'	onnxruntime_providers_mkldnn	c:\repos\pr3\onnxruntime\onnxruntime\core\providers\mkldnn\mkldnn_execution_provider.h	47	`

is something wrong in current implementation? thanks.

Reply:
@snnn. problem is that we will end up finding twice in original implementation. Implemented yours and returning `iter->second`. Thanks for point this out.